### PR TITLE
First enumerable

### PR DESCRIPTION
**Pull request by eduardosancho**
# What was done in this PR by @Sahar-AbdelSamad & @eduardosancho 🎃 : 

- [x] Create `my_enumerable.rb` module to mock the `all?`, `any?`, and `filter` methods.

- [x] Import the module into `my_list.rb` class that has an instance variable `@list`.

- [x] In `my_list.rb` implement a method #each that yields successive members of @list and uses the `my_enumerable.rb` module.

- [x] Test that the functionality of our mocked functions, matches the [real thing](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-all-3F).

rocky

### CODE REVIEW

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @eduardosancho and @Sahar-AbdelSamad, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**You have submitted a project, that does not have a descriptive README. Kindly add it**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+end
+
+list = MyList.new(1, 2, 3, 4)
+
+puts(list.all? { |a| a < 5 })
+puts(list.all? { |a| a > 5 })
+puts(list.any? { |e| e == 2 })
+puts(list.any? { |e| e == 5 })
+puts(list.filter(&:even?))
```
Comment (er-contreras) [inline my_list.rb]: ## OPTIONAL

- [x] I suggest using "p" method to see the real value behind what we really return. in this case would be  [2, 4]

**Review by er-contreras — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- Well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (Sahar-AbdelSamad) [reply my_list.rb]: For you for the valuable information :smile: We implemented it :+1:

**Pull request merged**